maat-court-data-api/src/main/java/gov/uk/courtdata/controller/ReservationsController.java

```diff
@@ -43,4 +51,39 @@ public ResponseEntity<Boolean> isMaatRecordLocked(
         log.info(String.format("Check if maatId is locked - %d {}", maatId));
         return ResponseEntity.ok(reservationsRepositoryHelper.isMaatRecordLocked(maatId));
     }
+
+
+    @GetMapping(value = "/recordname/{recordName}/recordid/{recordId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(description = "Get the reservation status object of a record")
+    @ApiResponse(responseCode = "200", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = AuthorizationResponse.class)))
+    @ApiResponse(responseCode = "400", description = "Bad Request.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ErrorDTO.class)))
+    @ApiResponse(responseCode = "500", description = "Server Error.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ErrorDTO.class)))
+    public ResponseEntity<ReservationsEntity> getReservationByRecordNameAndRecordId(
+            @PathVariable String recordName, @PathVariable Integer recordId,
+            @Parameter(description = "Used for tracing calls")
+            @RequestHeader(value = "Laa-Transaction-Id", required = false) String laaTransactionId) {
+        MDC.put(LoggingData.LAA_TRANSACTION_ID.getValue(), laaTransactionId);
+        log.info("Check reservation status - request received");
+        Optional<ReservationsEntity> reservationsEntity = reservationsRepositoryHelper.getReservationByRecordNameAndRecordId(recordName,recordId);
+        return reservationsEntity.map(c -> ResponseEntity.ok().body(c))
+                .orElse(ResponseEntity.notFound().build());
+    }
+
+
+
+    @GetMapping(value = "/username/{username}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(description = "Get the reservation status object of a record")
```

Comment:
Description could be more accurate, looks like it was copy pasted from previous endpoint.